rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,233 @@
+- Start Date: 31/03/2022
+
+# Table of contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Basic examples](#basic-examples)
+  - [Component Example - Controlled and Uncontrolled](#component-example---controlled-and-uncontrolled)
+  - [Component Example - Controlled](#component-example---controlled)
+- [Detailed design](#detailed-design)
+  - [Components](#components)
+- [Unit Testing](#unit-testing)  
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+- [Documentation](#documentation)
+- [Further Considerations](#further-considerations)
+
+# Summary
+
+The purpose of this document is to outline our intention to remove support for using Carbon input components in an uncontrolled manner.
+
+# Motivation
+
+- To reduce the code complexity in our input components. Currently we have to account for an input being uncontrolled or controlled and therefore have to test these both use cases. We would be able to remove any code relating to invariant messages regarding switching from controlled to uncontrolled or vice versa too.  
```

Comment:
I would say `Currently we have to account for an input being either uncontrolled or controlled` (ie add the word "either") to make it clearer that currently there are 2 cases we have to consider. Might just be me but it reads a bit strange to me without this.